contracts/token/ERC20/utils/SafeERC20.sol

```diff
@@ -35,24 +35,6 @@ library SafeERC20 {
         _callOptionalReturn(token, abi.encodeWithSelector(token.transferFrom.selector, from, to, value));
     }
 
-    /**
-     * @dev Deprecated. This function has issues similar to the ones found in
-     * {IERC20-approve}, and its usage is discouraged.
-     *
-     * Whenever possible, use {safeIncreaseAllowance} and
-     * {safeDecreaseAllowance} instead.
-     */
-    function safeApprove(IERC20 token, address spender, uint256 value) internal {
```

Comment:
Definitely another PR.

I'd keep the functions for sure, possibly using `forceApprove` under the hood. I remember this is a change I originally made that we reverted because it was breaking.

For renaming `forceApprove` into `safeApprove` ... I'm not sure. It could create confusion for people that migrate from 4.x to 5.x ... but at the same time I'm not sure this confusion is going to break anything (I'm convinced of the opposite, it will possibly avoid bugs)

Reply:
Decision was to use `forceApprove` for `safeIncreaseAllowance` and `safeDecreaseAllowance`, and not to rename `forceApprove` into `safeApprove` unless we are sure that there is no potential unsafety from bypassing the checks USDT has in approve.

Reply:
IMO that change should be another PR (its not a removal, its a refactor)

Reply:
Sounds good.

Reply:
#4260